### PR TITLE
added RecipeFilter interface and fixed tests

### PR DESCRIPTION
**Pull request by olensmar**

### CODE REVIEW

File: src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java [hunk 1]
```diff
@@ -37,6 +38,7 @@
     private final TestServerApi apiStub;
     private HttpBasicAuth authentication;
     private final List<ExecutionListener> executionListeners = new CopyOnWriteArrayList<>();
+    private final List<RecipeFilter> recipeFilters = new CopyOnWriteArrayList<>();
```
Comment (prakashmrec) [inline src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java]: Shouldn't we have a method to add RecipeFilter as well in this class?

File: src/test/java/com/smartbear/readyapi/client/DataSourceTestStepRecipeTest.java [hunk 2]
```diff
@@ -148,7 +150,13 @@ public void errorOccurred(Exception exception) {
                 assertThat(exception.getMessage(), is("Data source file not found: abc.xlsx"));
             }
         });
-        recipeExecutor.submitRecipe(recipe);
+
+        try {
+            recipeExecutor.submitRecipe(recipe);
+            assertTrue(false);
+        }
+        catch( com.smartbear.readyapi.client.execution.ApiException e ){
```
Comment (prakashmrec) [inline src/test/java/com/smartbear/readyapi/client/DataSourceTestStepRecipeTest.java]: formatting.

File: src/test/java/com/smartbear/readyapi/client/execution/RecipeExecutorTest.java [hunk 3]
```diff
@@ -141,8 +142,12 @@ public void throwsExceptionAndExecutionIsNullIfClientCertificateNotProvidedAndNo
         when(apiWrapper.postTestRecipe(eq(testRecipe.getTestCase()), eq(true), any(HttpBasicAuth.class))).thenReturn(pendingReport);
 
         executor.addExecutionListener(createExecutionListenerWithExpectedErrorMessage("Couldn't find client certificate file"));
-        Execution execution = executor.submitRecipe(testRecipe);
-        assertThat(execution, is(CoreMatchers.<Execution>nullValue()));
+        try {
+            Execution execution = executor.submitRecipe(testRecipe);
+            assertTrue(false);
+        }
+        catch( com.smartbear.readyapi.client.execution.ApiException e ){
```
Comment (prakashmrec) [inline src/test/java/com/smartbear/readyapi/client/execution/RecipeExecutorTest.java]: formatting

File: src/test/java/com/smartbear/readyapi/client/execution/RecipeExecutorTest.java [hunk 4]
```diff
@@ -157,8 +162,13 @@ public void throwsExceptionIfTestStepClientCertificateNotProvidedAndNotFoundOnSe
...
 
         executor.addExecutionListener(createExecutionListenerWithExpectedErrorMessage("Couldn't find test step client certificate file: clientCertificate.jks"));
-        Execution execution = executor.submitRecipe(testRecipe);
-        assertThat(execution, is(CoreMatchers.<Execution>nullValue()));
+
+        try {
+            executor.submitRecipe(testRecipe);
+            assertTrue(false);
+        }
+        catch( com.smartbear.readyapi.client.execution.ApiException e ){
```
Comment (prakashmrec) [inline src/test/java/com/smartbear/readyapi/client/execution/RecipeExecutorTest.java]: formatting

File: src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java (hunk 1 above)
Comment (olensmar) [inline src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java]: yes of course - thanks for noticing... :-)

**Comment by prakashmrec** [comment]
:+1:

**Pull request merged**